### PR TITLE
Np 1299 refactor to use simplified transformation

### PR DESCRIPTION
**Pull request by brinxmat**
- Simplify transformation
- Removed some values from the data (dates, these may need to be reinstated)
- Added testing for the various permutations given our current expectations (that anything goes)

### CODE REVIEW

**Review by svenngjeto — APPROVED** [review]

File: dynamodb-stream-reader/src/main/java/no/unit/nva/dynamodb/DynamoDBEventTransformer.java [hunk 1]
```diff
@@ -40,90 +36,87 @@ public DynamoDBEventTransformer() {
...
+    private String extractTitle(JsonNode record) {
+        return textFromNode(record, MAIN_TITLE_JSON_POINTER);
+    }
+
+    private String extractType(JsonNode record) {
+        return textFromNode(record, TYPE_JSON_POINTER);
+    }
+
+    private String formatDate(String year, String month, String day) {
+        return nonNull(year) && !year.isEmpty() ? String.format(FLAT_DATE_TEMPLATE, year, month, day) : null;
```
Comment (axthosarouris) [inline dynamodb-stream-reader/src/main/java/no/unit/nva/dynamodb/DynamoDBEventTransformer.java:96]: this returns 2020-null-null when month or day is null. Suggestion:

```
private String formatDate(String year, String month, String day) {

        if (isEmpty(day) && !isEmpty(month)) {
            return formatDateWithoutDay(year, month);
        } else if (isEmpty(month)) {
            return formatDateWithoutMonth(year);
        } else {
            return formatDateWithYearMonthDay(year, month, day);
        }
    }

private String formatDateWithYearMonthDay(String year, String month, String day) {
        LocalDate date = LocalDate.of(parseInt(year), parseInt(month), parseInt(day));
        return attempt(() -> date.format(DateTimeFormatter.ISO_LOCAL_DATE))
            .orElse(this::logAndReturnNull);
    }

    private boolean isEmpty(String input) {
        return isNull(input) || input.isBlank();
    }

    private String formatDateWithoutDay(String year, String month) {
        LocalDate date = LocalDate.of(parseInt(year), parseInt(month), IGNORED_VALUE);
        return date.format(DateTimeFormatter.ofPattern("yyyy-MM"));
    }

    private String formatDateWithoutMonth(String year) {
        LocalDate date = LocalDate.of(parseInt(year), IGNORED_VALUE, IGNORED_VALUE);
        return date.format(DateTimeFormatter.ofPattern("yyyy"));
    }
```

Comment (axthosarouris) [inline dynamodb-stream-reader/src/main/java/no/unit/nva/dynamodb/DynamoDBEventTransformer.java]: this returns <year>-null-null if month and day are null.

```
private String formatDate(String year, String month, String day) {
        if(isEmpty(year)){
            return null;
        }
        else if (isEmpty(day) && !isEmpty(month)) {
            return formatDateWithoutDay(year, month);
        } else if (isEmpty(month) ) {
            return formatDateWithoutMonth(year);
        } else {
            return formatDateWithYearMonthDay(year, month, day);
        }
    }

    private String formatDateWithYearMonthDay(String year, String month, String day) {
        LocalDate date = LocalDate.of(parseInt(year), parseInt(month), parseInt(day));
        return date.format(DateTimeFormatter.ISO_LOCAL_DATE);
    }

    private boolean isEmpty(String input) {
        return isNull(input) || input.isBlank();
    }

    private String formatDateWithoutDay(String year, String month) {
        LocalDate date = LocalDate.of(parseInt(year), parseInt(month), IGNORED_VALUE);
        return date.format(DateTimeFormatter.ofPattern("yyyy-MM"));
    }

    private String formatDateWithoutMonth(String year) {
        LocalDate date = LocalDate.of(parseInt(year), IGNORED_VALUE, IGNORED_VALUE);
        return date.format(DateTimeFormatter.ofPattern("yyyy"));
    }

private boolean isEmpty(String input) {
        return isNull(input) || input.isBlank();
    }
```

File: dynamodb-stream-reader/src/main/java/no/unit/nva/dynamodb/DynamoDBStreamHandler.java [hunk 2]
```diff
@@ -5,8 +5,8 @@
 import com.amazonaws.services.lambda.runtime.events.DynamodbEvent;
 import com.amazonaws.services.lambda.runtime.events.models.dynamodb.AttributeValue;
 import no.unit.nva.elasticsearch.Constants;
-import no.unit.nva.elasticsearch.ElasticSearchIndexDocument;
 import no.unit.nva.elasticsearch.ElasticSearchRestClient;
+import no.unit.nva.elasticsearch.IndexDocument;
```
Comment (axthosarouris) [inline dynamodb-stream-reader/src/main/java/no/unit/nva/dynamodb/DynamoDBStreamHandler.java:9]: Lines 31 and 32 have fields that are not in use.

File: dynamodb-stream-reader/src/main/java/no/unit/nva/dynamodb/DynamoDBEventTransformer.java (hunk 1 above)
Comment (axthosarouris) [reply dynamodb-stream-reader/src/main/java/no/unit/nva/dynamodb/DynamoDBEventTransformer.java]: Also should we log if the whole date is empty? Or is it expected many times to happen?

File: dynamodb-stream-reader/src/test/java/no/unit/nva/dynamodb/DynamoDBStreamHandlerTest.java [hunk 3]
```diff
@@ -143,50 +187,200 @@ public void testHandleExceptionInEventHandlingShouldGiveException() throws IOExc
     }
 
     @Test
-    public void dynamoDBStreamHandlerDoCreateHttpRequestFromModifyEvent() throws IOException, InterruptedException {
+    public void dynamoDBStreamHandlerCreatesHttpRequestWithIndexDocumentWithContributorsWhenInputIsModifyEvent() throws
```
Comment (axthosarouris) [inline dynamodb-stream-reader/src/test/java/no/unit/nva/dynamodb/DynamoDBStreamHandlerTest.java:196]:
```
DisplayName: dynamoDbSreamHandler creates request with index document with contributor when modify event contains contributors
```

File: dynamodb-stream-reader/src/test/java/no/unit/nva/dynamodb/DynamoDBStreamHandlerTest.java [hunk 4]
```diff
@@ -143,50 +187,200 @@ public void testHandleExceptionInEventHandlingShouldGiveException() throws IOExc
...
+        JsonNode requestBody = extractRequestBodyFromEvent(requestEvent);
+
+        IndexDocument expected = generateIndexDocument(identifier, contributorIdentifier, contributorName, mainTitle, type, date);
+        IndexDocument actual = mapper.convertValue(requestBody, IndexDocument.class);
+
+        assertThat(actual, samePropertyValuesAs(expected));
+    }
 
+    @Test
+    public void dynamoDBStreamHandlerCreatesHttpRequestWithIndexDocumentNullValuesWhenInputIsModifyEvent() throws
```
Comment (axthosarouris) [inline dynamodb-stream-reader/src/test/java/no/unit/nva/dynamodb/DynamoDBStreamHandlerTest.java:273]: write a display name indicating that input has null values.

**Review by axthosarouris — APPROVED** [review]

File: dynamodb-stream-reader/src/main/java/no/unit/nva/dynamodb/DynamoDBEventTransformer.java (hunk 1 above)
Comment (brinxmat) [reply dynamodb-stream-reader/src/main/java/no/unit/nva/dynamodb/DynamoDBEventTransformer.java]: The expectations are not clear, the date is not obligatory at the moment, we will address this in a subsequent PR

File: dynamodb-stream-reader/src/main/java/no/unit/nva/dynamodb/DynamoDBStreamHandler.java (hunk 2 above)
Comment (brinxmat) [reply dynamodb-stream-reader/src/main/java/no/unit/nva/dynamodb/DynamoDBStreamHandler.java:9]: This is addressed in another previous PR

**Review by axthosarouris — APPROVED** [review]

**Pull request merged**